src/trix/models/composition.coffee

```diff
@@ -76,17 +76,25 @@ class Trix.Composition extends Trix.BasicObject
     document = @document
     {index, offset} = document.locationFromPosition(position)
     block = document.getBlockAtIndex(index)
+    previousCharacter = block.getPreviousCharacter(offset)
+    nextCharacter = block.getNextCharacter(offset)
 
     if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
+      if previousCharacter is "\n"
+        document = document.removeTextAtRange([position - 1, position])
+      else if offset - 1 isnt 0
+        position += 1
```

Comment:
Why is this necessary? It isn't conditioned for the new block configs you've added so I imagine it could affect any block.